Firebase analytics - Dashboard

Sashen943 opened the pull request:
- Analytic functions added to presenter
- Implemented and injected through view controller
- Unit tests written for presenter 
- General code clean up of dashboard feature:
> Renaming variables
> Removed white spaces
> Added and improved MARK comments

Sashen943 committed:
Firebase analytics - Dashbaord

- Analytic functions added to presenter
- Implemented and injected through view controller
- Unit tests written for presenter 
- General code clean up of dashboard feature:
- Renaming variables
- Removed white spaces
- Added and improved MARK comments

pjwelcome commented on Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift:
```diff
@@ -33,33 +34,40 @@ class HomePresentableTests: XCTestCase {
+        presenterUnderTest = homePresenter
+    }
+
+    override func tearDown() {
```
Please remove the tear down if you are not using it for anything

Sashen943 replied:
done :)

pjwelcome commented on Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift:
```diff
@@ -33,33 +34,40 @@ class HomePresentableTests: XCTestCase {
         homePresenter.userDefaults = mockUserDefaults
-        systemUnderTest = homePresenter
+        homePresenter.analyticManager = mockAnalyticsManager
+        presenterUnderTest = homePresenter
```
why did you rename this to presenterUnderTest? systemUnder test is a more universal term for unit testing so that we know what the class we are testing. Please revert the renaming of systemUnderTest

Sashen943 replied:
done

Sashen943 committed:
Refactor code

- Renamed presenterUnderTest to systemUnderTest.
- Removed tearDown() method
- Addressed codacy and code climate issues of:
> lines being more than 120 characters
> Trailing white spaces

Sashen943 committed:
Codacy comment addressed

Sashen943 committed:
Screen did appear added

Sashen943 committed:
Removed forceful unwrapping
